services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -1,52 +1,337 @@
+  }
+
+  @Test
+  public void waypointIndicesList() {
+    List<Integer> indices = new ArrayList<>();
+    indices.add(1);
+    indices.add(5);
+    indices.add(7);
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointIndicesList(indices)
+        .build();
+
+    assertEquals(indices, routeOptions.waypointIndicesList());
+    assertEquals("1;5;7", routeOptions.waypointIndices());
+  }
+
+  @Test
+  public void waypointNamesString() {
+    String namesStr = "ab;;;cd;ef;;;gh;ij";
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointNames(namesStr)
+        .build();
+
+    assertEquals(namesStr, routeOptions.waypointNames());
+
+    List<String> names = routeOptions.waypointNamesList();
+    assertEquals("ab", names.get(0));
+    assertEquals(null, names.get(1));
+    assertEquals(null, names.get(2));
+    assertEquals("cd", names.get(3));
+    assertEquals("ef", names.get(4));
+    assertEquals(null, names.get(5));
+    assertEquals(null, names.get(6));
+    assertEquals("gh", names.get(7));
+    assertEquals("ij", names.get(8));
+  }
+
+  @Test
+  public void waypointNamesList() {
+    List<String> names = new ArrayList<>();
+    names.add(null);
+    names.add(null);
+    names.add("abc");
+    names.add(null);
+    names.add("def");
+    names.add(null);
+    names.add(null);
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointNamesList(names)
+        .build();
+
+    assertEquals(names, routeOptions.waypointNamesList());
+    assertEquals(";;abc;;def;;", routeOptions.waypointNames());
+  }
+
+  @Test
+  public void waypointTargetsString() {
+    String targetsStr = "1.2,3.4;;;5.65,7.123;;;";
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointTargets(targetsStr)
+        .build();
+
+    assertEquals("1.2,3.4;;;5.65,7.123", routeOptions.waypointTargets());
+
+    List<Point> targets = routeOptions.waypointTargetsList();
+    assertEquals(4, targets.size());
+    assertEquals(Point.fromLngLat(1.2, 3.4), targets.get(0));
+    assertEquals(null, targets.get(1));
+    assertEquals(null, targets.get(2));
+    assertEquals(Point.fromLngLat(5.65, 7.123), targets.get(3));
+  }
+
+  @Test
+  public void waypointTargetsList() {
+    List<Point> targets = new ArrayList<>();
+    targets.add(null);
+    targets.add(null);
+    targets.add(Point.fromLngLat(5.55, 7.77));
+    targets.add(Point.fromLngLat(1.22, 3.44));
+    targets.add(null);
+    targets.add(Point.fromLngLat(1.55, 8.99));
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointTargetsList(targets)
+        .build();
+
+    assertEquals(targets, routeOptions.waypointTargetsList());
+    assertEquals(";;5.55,7.77;1.22,3.44;;1.55,8.99", routeOptions.waypointTargets());
+  }
+
+  @Test
+  public void annotationsString() {
+    String annotationsStr = ANNOTATION_MAXSPEED + ";" + ANNOTATION_DURATION + ";" + ";";
```

Comment:
Same here. Not sure if we want to accept unnecessary `;` and then trim them 🤔 In any case, it's good for now 👍

Reply:
see the comment above